modin/test/exchange/dataframe_protocol/omnisci/test_protocol.py

```diff
@@ -0,0 +1,266 @@
+# Licensed to Modin Development Team under one or more contributor license agreements.
+# See the NOTICE file distributed with this work for additional information regarding
+# copyright ownership.  The Modin Development Team licenses this file to you under the
+# Apache License, Version 2.0 (the "License"); you may not use this file except in
+# compliance with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+
+"""Dataframe exchange protocol tests that are specific for OmniSci implementation."""
+
+import pytest
+import pyarrow as pa
+import pandas
+import numpy as np
+
+import modin.pandas as pd
+from modin.core.dataframe.pandas.exchange.dataframe_protocol.from_dataframe import (
+    convert_primitive_column_to_ndarray,
+    buffer_to_ndarray,
+    set_nulls,
+)
+from modin.pandas.utils import from_arrow, from_dataframe
+from modin.pandas.test.utils import df_equals
+from modin.test.test_utils import warns_that_defaulting_to_pandas
+from .utils import get_data_of_all_types, split_df_into_chunks, export_frame
+
+
+@pytest.mark.parametrize("data_has_nulls", [True, False])
+@pytest.mark.parametrize("from_omnisci", [True, False])
+def test_simple_export(data_has_nulls, from_omnisci):
+    if from_omnisci:
+        # OmniSci can't import 'uint64' as well as booleans
+        # issue for bool: https://github.com/modin-project/modin/issues/4299
+        exclude_dtypes = ["bool", "uint64"]
+    else:
+        exclude_dtypes = None
+
+    data = get_data_of_all_types(
+        has_nulls=data_has_nulls, exclude_dtypes=exclude_dtypes
+    )
+    md_df = pd.DataFrame(data)
+
+    exported_df = export_frame(md_df, from_omnisci)
+    df_equals(md_df, exported_df)
+
+    exported_df = export_frame(md_df, from_omnisci, nchunks=3)
+    df_equals(md_df, exported_df)
+
+    exported_df = export_frame(md_df, from_omnisci, nchunks=5)
+    df_equals(md_df, exported_df)
+
+    exported_df = export_frame(md_df, from_omnisci, nchunks=12)
+    df_equals(md_df, exported_df)
+
+
+@pytest.mark.parametrize("nchunks", [2, 4, 7])
+@pytest.mark.parametrize("data_has_nulls", [True, False])
+def test_export_aligned_at_chunks(nchunks, data_has_nulls):
+    """Test export from DataFrame exchange protocol when internal PyArrow table is equaly chunked."""
+    # Modin DataFrame constructor can't process PyArrow's category, so exclude it
```

Comment:
Why do we not exclude category dtype in the test above? How is that handled there?

Reply:
In other tests we build the modin frame like from a dictionary, the import flow is following:
1. data = dict
2. data = pandas.DataFrame(dict)
3. modin_frame.from_pandas(data)

There's no PyArrow in this flow, so no need to import arrow's dictionary type. In this particular test we build the modin frame directly from the PyArrow table using `from_arrow` function, which raises a non-implemented error on arrow dictionary type:
```python
>>> at = pyarrow.Table.from_pydict({"a": pandas.Categorical(["a", "a", "b"])})
>>> at.schema
a: dictionary<values=string, indices=int8, ordered=0>
>>> from_arrow(at)
NotImplementedError: dictionary<values=string, indices=int8, ordered=0>
```

Clarified this inside in-code comment to make it more clear:
```
# Modin DataFrame constructor can't process PyArrow's category when using ``from_arrow``, so exclude it
```

Reply:
ok